### DISCUSSION TITLE
Client Version too old

### DISCUSSION THREAD

**Discussion by Kevancic** [dandi/helpdesk#129] [Q&A] [answered]
I have a conda enviorment set up for dandi and I seem to be getting the following error. While in said Dandi conda enviorment I have rerun pip install dandi, and pip install --upgrade-dandi to no avail. I am unsure what I need to update to get this error away. Do I need to restart essentially and completely recreate my conda enviorment and redownload a new version of dandi? 

Error:
Error: Client version 0.51.0 is too old!  Server requires at least version 0.60.0

**Comment by yarikoptic** [comment]
Please `conda upgrade -c conda-forge dandi` , ref: https://anaconda.org/conda-forge/dandi shows 0.60.0 available

**Reply by Kevancic** [reply]
I ran both conda forge and pip install. When I ran Conda forge it did look like some updates did occur but when I run dandi upload I see this now 

Error output from dandi upload
`Error: Server requires schema version 0.6.7; client only supports 0.6.6.  You may need to upgrade dandi and/or dandischema.`

Output from conda upgrade -c conda-forge dandi (this is the second time of running it so below text not showing installed files I accidentally closed out of terminal with original run of command). 

```
conda upgrade -c conda-forge dandi
Collecting package metadata (current_repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.9.2
  latest version: 24.3.0

Please update conda by running

    $ conda update -n base -c defaults conda



# All requested packages already installed.
```

**Reply by CodyCBakerPhD** [reply]
Did you do this in a new fresh environment though?

Also what happens if you do `pip install -U dandischema`?

I ran into this error around the same time you did but I'm currently able to upload to dandi by upgrading both `dandi-cli` and `dandischema` in an previously existing environment

**Reply by Kevancic** [reply, marked as answer]
Running` pip install -U dandischema` did work and I have been able to start an upload!

**Reply by yarikoptic** [reply]
so it was installed via `pip` not `conda`.... that `conda` superpower makes it ambiguous on how to upgrade, heh

**Comment by yarikoptic** [comment]
Pypi also has even newer https://pypi.org/project/dandi/

if installed with pip, `pip install --upgrade dandi`. If doesn't work - provide cut paste of output from running that command
